### PR TITLE
Misc fixes: LPS-91247 LPS-91305 LPS-91310 LPS-91328 LPS-91348 LPS-91398 LPS-91404

### PR DESCRIPTION
**Pull request by ngriffin7a**

### CODE REVIEW

**Comment by ngriffin7a** [comment]
@shuyangzhou This PR contains fixes for various issues, one commit per issue. Each commit is very small and should be easy to review. Also, each issue has a reproducer portlet attached to the corresponding JIRA ticket in case you want to change something and want to do some testing.

If you are able to review this on Friday, March 1st that would be great because I hope to send backports to 7.1.x on Monday March 4th.

Thank you, Neil

**Comment by ngriffin7a** [comment]
ci:test

**Comment by ngriffin7a** [comment]
ci:retest

**Comment by ngriffin7a** [comment]
@shuyangzhou The files changed in this PR are all under modules/apps/bean-portlet so the failures reported by CI must be unrelated. Please review at your earliest convenience. Thanks, Neil

File: modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/scope/ScopedBeanManagerThreadLocal.java [hunk 1]
```diff
@@ -78,8 +78,12 @@ public static Closeable install(ScopedBeanManager scopedBeanManager) {
 			unsafeRunnable.run();
 		}
 		finally {
-			if (empty) {
-				scopedBeanManagers.pop();
+			if (!scopedBeanManagers.isEmpty()) {
```
Comment (shuyangzhou) [inline modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/scope/ScopedBeanManagerThreadLocal.java]: @ngriffin7a can scopedBeanManagers be empty here? You just made the push unconditional in the previous commit, won't that make it impossible for scopedBeanManagers to be empty here? Is any unsafeRunnable here messing with the threadlocal scopedBeanManagers ?

Comment (ngriffin7a) [reply modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/scope/ScopedBeanManagerThreadLocal.java]: @shuyangzhou Good catch. The check for `if (!scopedBeanManagers.isEmpty()) {` on line 81 is unnecessary. I removed it locally and tested the V3ResourceAsyncTests from the Portlet 3.0 TCK and it worked. I also checked in the debugger and the stack is always non-empty.

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/68919

**Pull request closed without merging**